lnrpc/chainrpc/chainnotifer_server.go

```diff
@@ -196,14 +196,26 @@ func (s *Server) RegisterWithRootServer(grpcServer *grpc.Server) error {
 func (s *Server) RegisterConfirmationsNtfn(in *ConfRequest,
 	confStream ChainNotifier_RegisterConfirmationsNtfnServer) error {
 
-	// We'll start by reconstructing the RPC request into what the
-	// underlying ChainNotifier expects.
+	// A height hint is required to prevent scanning the chain from genesis.
+	if in.HeightHint == 0 {
+		return errors.New("a height hint must be provided")
+	}
```

Comment:
Is this now true for all back ends? If it isn't, then we probably still want to enforce it here.